### PR TITLE
Module 1 has been completed.

### PR DESCRIPTION
**Pull request by srodriguez-lefebre**
Please check this and do questions

### CODE REVIEW

File: 1_WorkflowFundamentals/0_Pytorch model.py [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+step = 0.02
+X = torch.arange(start, end, step).unsqueeze(dim=1)
+y = weight * X + bias
+
+print(X[:10], y[:10])
+print(X.shape)
+print("\n")
+
+# Create train/test split
+train_split = int(0.8 * len(X)) # 80% of data used for training set, 20% for testing
```
Comment (JoaquinCampo) [inline 1_WorkflowFundamentals/0_Pytorch model.py:21]: why 80/20, why can we use another split?

File: 1_WorkflowFundamentals/0_Pytorch model.py [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+    plt.scatter(test_data, predictions, c="r", s=4, label="Predictions")
+
+  # Show the legend
+  plt.legend(prop={"size": 14});
+"""
+  
+  # Create a Linear Regression model class
+class LinearRegressionModel(nn.Module): # <- almost everything in PyTorch is a nn.Module (think of this as neural network lego blocks)
+    def __init__(self):
+        super().__init__()
```
Comment (JoaquinCampo) [inline 1_WorkflowFundamentals/0_Pytorch model.py:53]: What happens when we do super().__init__?

File: 1_WorkflowFundamentals/0_Pytorch model.py [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+model_0 = LinearRegressionModel()
+
+# Check the nn.Parameter(s) within the nn.Module subclass we created
+print(list(model_0.parameters()))
+
+# List named parameters 
+print(model_0.state_dict())
+
+# Make predictions with model
+with torch.inference_mode():
```
Comment (JoaquinCampo) [inline 1_WorkflowFundamentals/0_Pytorch model.py:79]: what is this line doing?

File: 1_WorkflowFundamentals/0_Pytorch model.py [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+# with torch.no_grad():
+#   y_preds = model_0(X_test)
+
+# Check the predictions
+print(f"Number of testing samples: {len(X_test)}") 
+print(f"Number of predictions made: {len(y_preds)}")
+print(f"Predicted values:\n{y_preds}")
+
+# Create the loss function
+loss_fn = nn.L1Loss() # MAE loss is same as L1Loss
```
Comment (JoaquinCampo) [inline 1_WorkflowFundamentals/0_Pytorch model.py:92]: what is MAE?

File: 1_WorkflowFundamentals/0_Pytorch model.py [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+
+# Create the optimizer
+optimizer = torch.optim.SGD(params=model_0.parameters(), # parameters of target model to optimize
```
Comment (JoaquinCampo) [inline 1_WorkflowFundamentals/0_Pytorch model.py:95]: We are already familizar with Gradient Descent, GD for short, but what is SGD? What is the difference to GD?

File: 1_WorkflowFundamentals/0_Pytorch model.py [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+epoch_count = []
+
+for epoch in range(epochs):
+    ### Training
+
+    # Put model in training mode (this is the default state of a model)
+    model_0.train()
+
+    # 1. Forward pass on train data using the forward() method inside 
+    y_pred = model_0(X_train)
```
Comment (JoaquinCampo) [inline 1_WorkflowFundamentals/0_Pytorch model.py:116]: what happens when we do model(x)? Hint: Research about the \_\_call__ method. Maybe you can look into the implementation of some open source machine learning libraries, e.g. nn.module

File: 1_WorkflowFundamentals/0_Pytorch model.py [hunk 7]
```diff
@@ -0,0 +1,200 @@
...
+    # print(y_pred)
+
+    # 2. Calculate the loss (how different are our models predictions to the ground truth)
+    loss = loss_fn(y_pred, y_train)
+
+    # 3. Zero grad of the optimizer
+    optimizer.zero_grad()
+
+    # 4. Loss backwards
+    loss.backward()
```
Comment (JoaquinCampo) [inline 1_WorkflowFundamentals/0_Pytorch model.py:126]: This is an interesting line, a lot is happening when we do backward(). Its not mandatory to understand it now, but bare in mind that there is a whole process of gradient calculations going on here

File: 1_WorkflowFundamentals/0_Pytorch model.py [hunk 8]
```diff
@@ -0,0 +1,200 @@
...
+
+    # 5. Progress the optimizer
+    optimizer.step()
```
Comment (JoaquinCampo) [inline 1_WorkflowFundamentals/0_Pytorch model.py:129]: What is this step? What goes on in the background here?

**Review by JoaquinCampo — APPROVED** [review]
